src/Connection.php

```diff
@@ -214,8 +214,10 @@ public function cursor($query, $bindings = [], $useReadPdo = false)
             yield $result;
         }
 
-        if ($limit > $numResults) {
-            foreach ($this->scroll($scrollId, $scrollTimeout, $limit - $numResults) as $result) {
+        if (! $limit || $limit > $numResults) {
+            $limit = $limit ? $limit - $numResults : $limit;
```

Comment:
I think I'm being an idiot, but I'm not sure I understand this line, I mean I get that you're setting the limit, but I can figure out what `$limit - $numResults` is doing. So I have a limit of, say, `100`, and the number of results is `25`, set the limit to `75`?

Reply:
Sorry yeah, it's confusing. The default limit will be zero so all results will be returned, but if you want a scroll cursor with 1000 results and the first query (on line 205) returns 100 results, then you want to scroll through 900 additional results, so 900 gets passed to `scroll()` and at line 255 it will break out of the scroll once that's been hit.

Reply:
Ohhh I see, ok I'm with you!